### PR TITLE
refactor!: Migrate Ktor feature to new plugin API

### PR DESCRIPTION
**Pull request by stuebingerb**
Migrates the Ktor feature to the [new plugin API](https://ktor.io/docs/server-custom-plugins.html) to hopefully support 2.x and 3.x.

Potentially breaking: the return type of `install(GraphQL)` changes from `GraphQL` to `ApplicationPlugin<Configuration>`

### CODE REVIEW

File: kgraphql-ktor/src/test/kotlin/com/apurebase/kgraphql/KtorConfigurationTest.kt [hunk 1]
```diff
@@ -1,38 +0,0 @@
...
-class KtorConfigurationTest : KtorTest() {
-
-    @Test
-    fun `default configuration should`() {
-        var checked = false
-        testApplication {
-            application {
-                val config = install(GraphQL) {}
-                checked = true
-                config.schema.configuration.executor shouldBeEqualTo Executor.Parallel
```
Comment (stuebingerb) [inline kgraphql-ktor/src/test/kotlin/com/apurebase/kgraphql/KtorConfigurationTest.kt:18]: Not sure what exactly this was supposed to test but as the return type of `install(GraphQL)` has changed the schema is no longer available. Hopefully this is not a problem 🤔

**Comment by stuebingerb** [comment]
Thinking about replacing the feature with dedicated routes in the (not so far) future, similar to https://github.com/aPureBase/KGraphQL/issues/190

**Review by mervyn-mccreight — COMMENTED** [review]
I think if we plan to try to support both, 2.x and 3.x we'd need to change the dependency of ktor-server-core from implementation to compileOnly.
Otherwise we would force every user into using 3.x by default AFAIK.

**Comment by mervyn-mccreight** [comment]
> Thinking about replacing the feature with dedicated routes in the (not so far) future, similar to https://github.com/aPureBase/KGraphQL/issues/190

I had this thought as well :)

**Review by mervyn-mccreight — APPROVED** [review]

**Comment by stuebingerb** [comment]
Unfortunately, the new plugin API is also not compatible with Ktor 2:
```
Exception in thread "main" java.lang.NoClassDefFoundError: io/ktor/server/routing/RoutingRootKt
	at com.apurebase.kgraphql.KtorFeatureKt.createPlugin$lambda$4(KtorFeature.kt:51)
```
I assume this is due to the same issue the original plugin was not compatible, namely the refactoring of `Routing` to `RoutingRoot` in Ktor 3. I would then re-use #26 and (only) support Ktor 3 for now. And then let's get rid of the feature afterwards.

**Pull request closed without merging**